### PR TITLE
Beam geometry

### PR DESCRIPTION
**Pull request by chenkasirer**
- Restored the `geometry` accessor property to simplify Beam API
- Geometry is now created on first call to `beam.geometry` rather than automatically. This allows creating a `TimberAssembly` in contexts where geometry is not available or not of interest (e.g. cadwork).

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_timber/ghpython/components/CT_ShowAssembly/code.py [hunk 1]
```diff
@@ -12,6 +12,6 @@ def RunScript(self, Assembly):
 
         Brep = []
         for beam in Assembly.beams:
-            Brep.append(Artist(beam.get_geometry(True)).draw())
+            Brep.append(Artist(beam.geometry).draw())
```
Comment (funkchaser) [inline src/compas_timber/ghpython/components/CT_ShowAssembly/code.py:15]: check also other GH components, there are some more `beam.get_geometry` calls

**Review by funkchaser — CHANGES_REQUESTED** [review]
Looks otherwise good to me, looks simpler/neater, but haven't run any tests/demos with the new workflow. 
Can you also check if it requires updating the documentation/tutorial?

File: src/compas_timber/ghpython/components/CT_ShowAssembly/code.py (hunk 1 above)
Comment (chenkasirer) [reply src/compas_timber/ghpython/components/CT_ShowAssembly/code.py:15]: Nice catch :) Fixed in Bake component. Seen anywhere else?

**Comment by chenkasirer** [comment]
> n a

Updated `Beam`'s docstring. Couldn't find any references in the docs. However, `get_geometry` still shows as method of `Beam` (inherited from `Part`) I would have to sort that out in COMPAS core at some point.

BTW, since `compas_timber` continues to rely on unreleased changes to COMPAS, I added the exact commit I'm currently based on to `requriements.txt` and in order to test you'd need to install using `pip install -r requirements-dev.txt`. Maybe better in a new env so to not mess up any other working projects that use COMPAS LTS.

**Comment by chenkasirer** [comment]
@funkchaser ping :)

File: src/compas_timber/parts/beam.py [hunk 2]
```diff
@@ -435,6 +416,10 @@ def from_endpoints(cls, point_start, point_end, width, height, z_vector=None, ge
         line = Line(point_start, point_end)
         return cls.from_centerline(line, width, height, z_vector, geometry_type)
 
+    @classmethod
+    def from_brep(cls, brep):
+        pass
```
Comment (gonzalocasas) [inline src/compas_timber/parts/beam.py:421]: maybe `raise NotImplementedError` instead of passing?

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,4 +1,3 @@
-black
```
Comment (gonzalocasas) [inline requirements-dev.txt:1]: no `black`?

File: src/compas_timber/connections/t_butt.py [hunk 4]
```diff
@@ -82,10 +82,10 @@ def cutting_plane(self):
         cfr = Frame(cfr.point, cfr.yaxis, cfr.xaxis)  # flip normal towards the inside of main beam
         return cfr
 
-    def restore_beams_from_keys(self, assemly):
+    def restore_beams_from_keys(self, assembly):  # TODO: fix typo
```
Comment (gonzalocasas) [inline src/compas_timber/connections/t_butt.py:85]: the typo seems fixed, I guess the comment can go?

**Review by gonzalocasas — APPROVED** [review]
LGTM!

File: src/compas_timber/parts/beam.py (hunk 2 above)
Comment (chenkasirer) [reply src/compas_timber/parts/beam.py]: removed altogether. will be in it's own PR.

File: requirements-dev.txt (hunk 3 above)
Comment (chenkasirer) [reply requirements-dev.txt:1]: was there twice! :0

**Pull request merged**